### PR TITLE
Add extra configuration options: appVersion, environment and rootDirectory

### PR DESCRIPTION
**Pull request by aminin**

### CODE REVIEW

File: src/Notifier.php [hunk 1]
```diff
@@ -105,9 +111,13 @@ public function buildNotice($exc)
             ],
             'os' => php_uname(),
             'language' => 'php '.phpversion(),
+            'rootDirectory' => $this->opt['rootDirectory'] ?: $_SERVER['DOCUMENT_ROOT'] ?: null,
```
Comment (vmihailenco) [inline src/Notifier.php]: Can you please remove double `?:` usage? It is not obvious what is the result of this statement...

File: src/Notifier.php [hunk 2]
```diff
@@ -41,7 +44,10 @@ public function __construct($opt = [])
 
         $this->opt = array_merge([
             'host' => 'api.airbrake.io',
+            'rootDirectory' => null,
```
Comment (vmihailenco) [inline src/Notifier.php]: Is this needed? It catches attention because this is the only option with null value.

File: src/Notifier.php [hunk 3]
```diff
@@ -226,4 +236,22 @@ protected function getNoticesURL()
...
+                        continue;
+                    }
+                    foreach ($error['backtrace'] as &$frame) {
+                        $frame['file'] = preg_replace("~^$projectRoot~", '[PROJECT_ROOT]', $frame['file']);
+                    }
+                }
+            }
+            return $notice;
+        });
+    }
```
Comment (vmihailenco) [inline src/Notifier.php:257]: I suggest to create `rootDirectoryFilter`:

```
protected function rootDirectoryFilter()
{
    foreach ($notice['errors'] as &$error) {
        ...
    }
}
```

and then add it like this in constructor:

```
if (!empty($this->opt['rootDirectory']) {
    $this->addFilter($this->rootDirectoryFilter);
}
```

This way we can write a test for this filter.

**Comment by vmihailenco** [comment]
Looks good, but can you please add a simple test for `rootDirectory` option?

File: src/Notifier.php (hunk 1 above)
Comment (aminin) [inline src/Notifier.php]: It looks as clear as `foo = a || b || 'default'` in ruby.

Yet `$_SERVER['DOCUMENT_ROOT']` is bad default value for the `rootDirectory` option.
1. In CLI it has no meaning.
2. Airbrake users hardly ever store their code in DOCUMENT_ROOT

**Comment by aminin** [comment]
@vmihailenco I'm done.

**Comment by vmihailenco** [comment]
Thanks.

**Pull request merged**